text/0005-application-formalization.md

```diff
@@ -0,0 +1,191 @@
+- Start Date: 10/09/2023
+- Contributors:
+  - [Craig McNally](cmcnally@ebsco.com)
+  - [Vince Bareau](vbareau@ebsco.com)
+- RFC PRs:
+  - DRAFT REVIEW: https://github.com/folio-org/rfcs/pull/14
+  - PRELIMINARY REVIEW: TBD
+  - PUBLIC REVIEW: TBD
+  - FINAL REVIEW: TBD
+- Outcome: <!-- Leave this blank. Will eventually be either ACCEPTED or REJECTED -->
+
+# Application Formalization
+
+## Summary
+It is proposed that an Application be defined as the minimal but complete set of elements which together are intended to deliver a specific solution to Folio. The intention is to group together all the closely related components which will need to evolve together, but independently of other parts (i.e. other applications) of Folio.
```

Comment:
I think it could be useful to further describe what folks mean by complete in this context

For example, `circulation` activities require access to `users` to be completed, that would suggest that `users` would have to be part of the `circulation` app for it to be complete

Reply:
I have what I think is a related question - what is the rubric for deciding whether something should share a database or whether the interaction should rely on API calls? This is the difference for whether things are part of the same application or not, right?

Reply:
In the `circulation` vs `users` example, my understanding is that those would be two separate applications, thought `circulation` would depend on `users` being installed and available. `users` is responsible for managing users, `circulation` is responsible for managing circulation transactions (perhaps as narrow as check-outs and check-ins), but would need to reference both `users` and `inventory`.

Reply:
> circulation is responsible for managing circulation transactions (perhaps as narrow as check-outs and check-ins), but would need to reference both users and inventory

That's why I'm interested in what `complete` means because circulation cannot currently operate without those dependencies

Reply:
@starsplatter regarding:
> what is the rubric for deciding whether something should share a database or whether the interaction should rely on API calls

It sounds like you're conflating the bounded context and application formalization RFCs.  To be fair, I think your question predated the bounded context RFC abstract, so the distinction might have been less clear at the time.

Reply:
It's also hard to discuss application formalization without bounded context.  See my suggestion about [reordering](https://github.com/folio-org/rfcs/pull/20/files#r1394160071).

Reply:
> The use of "complete" in this context means to me that an application should include not only backend modules, but also corresponding ui module(s), etc

Ok. I think that definition should be made explicit within the RFC

> Also, keep in mind that defining applications is not a "one and done" activity. Applications composition will undoubtedly change over time

Agreed. My interest is in the language we use during the conversations about (the idea / intent / design of) applications and as guidance for decisions about the boundaries between specific applications going forward

Reply:
I think this has been addressed.  If needed, we can add to and refine the terminology section during later phases of the RFC process.